File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        var i = 0;
        var j = nums.length - 1;
        while (i <= j) {
            var mid = ((j - i) / 2) + i;
            if (nums[mid] > target) {
                j = mid - 1;
            } else if (nums[mid] < target) {
                i = mid + 1;
            } else {
                return mid;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-1.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        nums = []
        i = 0
        j = 0

        while i < len(nums1) or j < len(nums2):
            if i >= len(nums1):
                nums.append(nums2[j])
                j += 1
            elif j >= len(nums2) or nums1[i] < nums2[j]:
                nums.append(nums1[i])
                i += 1
            elif nums2[j] < nums1[i]:
                nums.append(nums2[j])
                j += 1
            else:
                nums.append(nums1[i])
                i += 1
        
        mid = len(nums) // 2

        if len(nums) % 2 == 0:
            return (nums[mid] + nums[mid - 1]) / 2
        else:
            return nums[mid]
        